src/RelationParser.h

```diff
@@ -0,0 +1,74 @@
+//
+//  RelationParser.h
+//  snowcrash
+//
+//  Created by Pavan Kumar Sunkara on 04/03/15.
+//  Copyright (c) 2015 Apiary Inc. All rights reserved.
+//
+
+#ifndef SNOWCRASH_RELATIONPARSER_H
+#define SNOWCRASH_RELATIONPARSER_H
+
+#include "SectionParser.h"
+#include "RegexMatch.h"
+
+namespace snowcrash {
+
+    /** Link Relation matching regex */
+    const char* const RelationRegex = "^[[:blank:]]*[Rr]elation[[:blank:]]*:" SYMBOL_IDENTIFIER "?$";
+
+    /**
+     *  Relation Section Processor
+     */
+    template<>
+    struct SectionProcessor<Relation> : public SectionProcessorBase<Relation> {
```

Comment:
I think we need to do something about this `SectionProcessor`. If I have

```
typedef std::string Relation;
typedef std::string Asset;
```

Then, defining `SectionProcessor<Asset>` and `SectionProcessor<Relation>` gives me redefinition errors. Which is why I had to do `struct Relation` earlier.